internal/pkg/views/list.go

```diff
@@ -219,19 +233,32 @@ func (w *ListWidget) Refresh() {
 	}
 
 	w.GoBack()
-	w.ExpandCurrentSelection()
+	w.expandItem(currentExpandedItem)
 
-	w.ChangeSelection(currentSelection)
+	// wait for navigation before resetting previous selection
+	go func() {
+		navigatedChannel := eventing.SubscribeToTopic("list.navigated")
```

Comment:
```suggestion
		defer errorhandling.RecoveryWithCleanup()
		navigatedChannel := eventing.SubscribeToTopic("list.navigated")
```

In the off chance that this fails adding recovery will ensure the terminal is left in a usable state and stack is printed etc

Reply:
Think this'll need an additional import of `"github.com/lawrencegripper/azbrowse/internal/pkg/errorhandling"`